### ISSUE TITLE
Unresolvable requirements file

### ISSUE THREAD

**Issue by tompollard** [chorus-ai/chorus_waveform#103]
Tests are currently failing because the dependencies in requirements.txt cannot be resolved. e.g. see: https://github.com/chorus-ai/chorus_waveform/actions/runs/11406512755/job/31740432168

```
 1 warning found (use docker --debug to expand):
 - FromAsCasing: 'as' and 'FROM' keywords' casing do not match (line 1)
Dockerfile:23
--------------------
  21 |                     flake8 \
  22 |                     pytest
  23 | >>> RUN pip install --no-cache-dir -r requirements.txt
  24 |     RUN pip install --no-cache-dir --no-deps -e .
  25 |     
--------------------
ERROR: failed to solve: process "/bin/sh -c pip install --no-cache-dir -r requirements.txt" did not complete successfully: exit code: 1
Error: buildx failed with: ERROR: failed to solve: process "/bin/sh -c pip install --no-cache-dir -r requirements.txt" did not complete successfully: exit code: 1
```

Most likely the cause is that wfdb-python is currently incompatible with Numpy V2 (see: https://github.com/MIT-LCP/wfdb-python/issues/493).

**Comment by tompollard** [comment]
I think I'm unfairly blaming WFDB. According to the [error here](https://github.com/chorus-ai/chorus_waveform/actions/runs/11443416337/job/31836008103), we are requiring a version of AtriumDB that doesn't exist:

https://github.com/chorus-ai/chorus_waveform/blob/426cbc82eae35270ab103f4efab6fee91c2803aa/requirements.txt#L1

```
> pip index versions atriumdb

atriumdb (2.3.0)
Available versions: 2.3.0, 2.2.3, 2.2.2, 2.2.1, 2.2.0, 2.1.1, 2.1.0, 2.0.0
```

**Comment by briangow** [comment]
@tompollard , yes, this is expected. @WilliamDixon provided an early version of 2.4 and the code that depends on that has been merged. I believe the new package should be released publicly very soon.

**Comment by tompollard** [comment]
I see, in this case the requirements should be updated to use the new version (e.g. to pip install from `pip install git+https://github.com/LaussenLabs/atriumdb.git@e860b30cfb93a2a290ac120c1c09a2af7806fb2b`).

**Comment by WilliamDixon** [comment]
@tompollard @briangow I have a new PR up that works with the newly release atriumdb 2.4.0 and have updated the requirements.txt to reflect that.

**Comment by WilliamDixon** [comment]
> I see, in this case the requirements should be updated to use the new version (e.g. to pip install from `pip install git+https://github.com/LaussenLabs/atriumdb.git@e860b30cfb93a2a290ac120c1c09a2af7806fb2b`).

In the future if I use a PR that depends on pre-released features, I'll be sure to use syntax like this so its available for everyone automatically.
